File: v3/cmd/signoffautomator/signoffautomator.js
const SignoffAutomatorApi = require('../../pkg/signoffautomatorapi/signoffautomatorapi.js')

let main = async function() {
    // Main execution
    try {
        var api = new SignoffAutomatorApi()
    } catch ({ message }) {
        console.log("Error when initializing api", message)
        process.exit(1)
    }
    try {
        await api.execute()
    } catch ({ message }) {
        console.log("Error when Executing api", message)
    }
    console.log("Main execution done");
    if (SignoffAutomatorApi.getDebug()) {
        SignoffAutomatorApi.resetDebug();
    }
}

main();

File: src/main/java/io/github/johnnylina/signoffautomator/SignoffAutomator.java
package io.github.johnnylina.signoffautomator;

public class SignoffAutomator {
    public static void main(String[] args) {
        final SignoffAutomatorApi api;
        try {
            api = new SignoffAutomatorApi();
        } catch (IllegalArgumentException iae) {
            System.out.println("Error when initializing api" + iae.getMessage());
            System.exit(1);
            return;
        }
        try {
            api.execute();
        } catch (RuntimeException re) {
            System.out.println("Error when Executing api " + re.getMessage());
            if (SignoffAutomatorApi.getDebug()) {
                re.printStackTrace();
            }
        }
        System.out.println("Main execution done");
        if (SignoffAutomatorApi.getDebug()) {
            SignoffAutomatorApi.resetDebug();
        }
    }
}